app-config-node/src/file-source.ts

```diff
@@ -46,8 +47,8 @@ export class FlexibleFileSource extends ConfigSource {
     private readonly filePath: string,
     private readonly environmentOverride?: string,
     private readonly environmentAliases: EnvironmentAliases = defaultAliases,
-    private readonly fileExtensions: string[] = ['yml', 'yaml', 'toml', 'json', 'json5'],
     private readonly environmentSourceNames?: string[] | string,
+    private readonly fileExtensions: string[] = ['yml', 'yaml', 'toml', 'json', 'json5'],
```

Comment:
Constructor args changed order - maybe use the trio type to avoid a footgun here